packages/generators/init-generator.ts

```diff
@@ -28,274 +19,212 @@ import { WebpackOptions } from "./types";
-			Object.keys(entry).forEach((name: string): void => generateEntryFile(entry[name], `${name} main file!`));
-		}
-
-		// Generate README
-		this.fs.copyTpl(path.resolve(__dirname, "./templates/README.md"), this.destinationPath("README.md"), {});
-
-		// Generate HTML template file
-		if (this.usingDefaults) {
-			this.fs.copyTpl(
-				path.resolve(__dirname, "./templates/template.html"),
-				this.destinationPath("index.html"),
-				{}
-			);
-		}
-
-		// Genrate tsconfig
-		if (this.langType === LangType.Typescript) {
-			const tsConfigTemplatePath = "./templates/tsconfig.json.js";
-			this.fs.extendJSON(this.destinationPath("tsconfig.json"), require(tsConfigTemplatePath));
-		}
-	}
+        // Chunksplitting
+        this.configuration.config.webpackOptions.optimization = getDefaultOptimization(this.usingDefaults);
+        this.configuration.config.webpackOptions.mode = this.usingDefaults ? "'production'" : "'development'";
+        done();
+    }
+
+    public installPlugins(): void {
+        const packager = getPackageManager();
+        const opts: {
+            dev?: boolean;
+            'save-dev'?: boolean;
+        } = packager === 'yarn' ? { dev: true } : { 'save-dev': true };
+
+        this.scheduleInstallTask(packager, this.dependencies, opts);
+    }
+
+    public writing(): void {
+        this.configuration.usingDefaults = this.usingDefaults;
+        this.config.set('configuration', this.configuration);
+
+        if (this.langType === 'ES6') {
+            this.fs.copyTpl(path.resolve(__dirname, './templates/.babelrc'), this.destinationPath('.babelrc'), {});
+        }
+        const packageJsonTemplatePath = './templates/package.json.js';
+        this.fs.extendJSON(this.destinationPath('package.json'), require(packageJsonTemplatePath)(this.usingDefaults));
+
+        const generateEntryFile = (entryPath: string, name: string): void => {
+            entryPath = entryPath.replace(/'/g, '');
+            this.fs.copyTpl(path.resolve(__dirname, './templates/index.js'), this.destinationPath(entryPath), { name });
+        };
+
+        // Generate entry file/files
+        const entry = this.configuration.config.webpackOptions.entry || './src/index.js';
+        if (typeof entry === 'string') {
+            generateEntryFile(entry, 'your main file!');
+        } else if (typeof entry === 'object') {
+            Object.keys(entry).forEach((name: string): void => generateEntryFile(entry[name], `${name} main file!`));
+        }
+
+        // Generate README
+        this.fs.copyTpl(path.resolve(__dirname, './templates/README.md'), this.destinationPath('README.md'), {});
+
+        // Generate HTML template file, copy default service worker
+        if (this.usingDefaults) {
+            this.fs.copyTpl(path.resolve(__dirname, './templates/template.html'), this.destinationPath('index.html'), {});
+            this.fs.copyTpl(path.resolve(__dirname, './templates/sw.js'), this.destinationPath('sw.js'), {});
```

Comment:
Copied over the sw file as well on default scaffold, this was already imported in the template html file but the sw file was absent so added it.